src/IntaRNA/Accessibility.cpp

```diff
@@ -114,4 +114,71 @@ writeRNAplfold_text( std::ostream& out, const E_type RT, const bool writeProbs )
 
 ////////////////////////////////////////////////////////////////////
 
+void
+Accessibility::
+compute_accessible_ranges( IndexRangeList& ranges, const size_t max_seq_length) const
+{
+	if(getSequence().size() < max_seq_length){
+		ranges.push_back(IndexRange(0,getSequence().size()-1));
+	}
+	else{
+		compute_accessible_ranges_recursive( ranges, max_seq_length, ED_UPPER_BOUND, 0, getSequence().size());
+	}
+}
+
+////////////////////////////////////////////////////////////////////
+
+void
+Accessibility::
+compute_accessible_ranges_recursive( IndexRangeList& ranges, const size_t max_seq_length, E_type threshold, int i, int j) const
+{
+
+	const int l = 50; // Arbitrarily chosen window
+	int from = -1;
+	int to = -1;
+	E_type ED = ED_UPPER_BOUND;
+
+	for (i; i<j+1-l; i++) {
```

Comment:
this setup prevents splits eg at `i+l` if `j=i+(1.5*l)`, sinc i runs only up to `i+(0.5*l)`

ie, you do not consider all valid splits.. 

if not interested, you can also initialize the running i with initiali+l